sqlglot/generator.py

```diff
@@ -1276,11 +1276,19 @@ def bitstring_sql(self, expression: exp.BitString) -> str:
             return f"{self.dialect.BIT_START}{this}{self.dialect.BIT_END}"
         return f"{int(this, 2)}"
 
-    def hexstring_sql(self, expression: exp.HexString) -> str:
+    def hexstring_sql(self, expression: exp.HexString, binary_function_repr: bool = False) -> str:
         this = self.sql(expression, "this")
-        if self.dialect.HEX_START:
-            return f"{self.dialect.HEX_START}{this}{self.dialect.HEX_END}"
-        return f"{int(this, 16)}"
+        is_integer_type = expression.args.get("is_integer")
+
+        if is_integer_type or (not self.dialect.HEX_START and not binary_function_repr):
+            return f"{int(this, 16)}"
+        elif not is_integer_type and self.dialect.HEX_STRING_IS_INTEGER_TYPE:
```

Comment:
Reaching the second branch means `is_integer_type` must be `False`, otherwise we'd take the 1st one:

```suggestion
        if self.dialect.HEX_STRING_IS_INTEGER_TYPE:
```